logic/src/commonMain/kotlin/com/wire/kalium/logic/data/call/CallRepository.kt

```diff
@@ -139,124 +138,227 @@ internal class CallDataSource(
             establishedTime = null
         )
 
-        val callProfile = _callProfile.value
-        val updatedCalls = callProfile.calls.toMutableMap().apply {
-            this[call.conversationId.toString()] = call
+        val isCallInCurrentSession = _callMetadataProfile.value.data.containsKey(conversationId.toString())
+        val lastCallStatus = callDAO.getCallStatusByConversationId(conversationId = callEntity.conversationId)
+        val isGroupCall = callEntity.conversationType == ConversationEntity.Type.GROUP
+        val activeCallStatus = listOf(
+            CallEntity.Status.ESTABLISHED,
+            CallEntity.Status.ANSWERED,
+            CallEntity.Status.STILL_ONGOING
+        )
+
+        if (status == CallStatus.INCOMING) {
```

Comment:
So, the thought on it was moving everything to its respective use cases (incoming call, start call use case and etc.) but some of the code FOR NOW would be duplicated because both places would need the call entity, metadata and other info for the logic check.

I have decided to keep it in the repository as for now so we can move forward with calling and re-visit the topic on rearranging later on. what do you think?

Reply:
makes sense to me! we can revisit later on 🤝 🙌